### PR TITLE
GenericMapStore tests for non-SQL backends [HZ-1635]

### PR DESCRIPTION
**Pull request by TomaszGaweda**
Jira: https://hazelcast.atlassian.net/browse/HZ-1635

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [ ] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by TomaszGaweda** [comment]
run-nightly-tests

**Comment by TomaszGaweda** [comment]
run-nightly-tests

**Comment by TomaszGaweda** [comment]
run-nightly-tests

**Comment by TomaszGaweda** [comment]
^ nightly test failures seems not connected to this PR

**Comment by TomaszGaweda** [comment]
run-nightly-tests

**Comment by TomaszGaweda** [comment]
run-nightly-tests

File: hazelcast/src/test/java/com/hazelcast/test/jdbc/GenericMapStoreObjectProvider.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ */
+package com.hazelcast.test.jdbc;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Allows creating of objects (tables, collections) used in GenericMapStore tests.
+ */
+public interface GenericMapStoreObjectProvider extends TestDatabaseProvider {
```
Comment (frant-hartm) [inline hazelcast/src/test/java/com/hazelcast/test/jdbc/GenericMapStoreObjectProvider.java:24]: I like the idea that we would have something that is able to create and verify objects in the native database, but I am not sure if we should tie it with the `DatabaseProvider`. Maybe the GenericMapStoreObjectProvider can use the database provider, but it should be a separate object.

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapLoaderTest.java [hunk 2]
```diff
@@ -75,6 +82,8 @@ public static void beforeClass() {
     public void setUp() {
         hz = instances()[0];
         mapName = "people_" + randomName();
+        assertThat(databaseProvider).isInstanceOf(GenericMapStoreObjectProvider.class);
+        objectProvider = (GenericMapStoreObjectProvider) databaseProvider;
```
Comment (frant-hartm) [inline extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapLoaderTest.java:86]: So now that we have the `GenericMapStoreObjectProvider` could we have another parent class - not `JdbcSqlTestSupport`? (which is specific to Jdbc, but we use something else for Mongo).

File: hazelcast/src/test/java/com/hazelcast/test/jdbc/GenericMapStoreObjectProvider.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * Creates a database object (table, collection, etc) with two fields:
+     * - id or {@code person-id}, depending on {@code #useQuotedNames} option
+     * - name
+     */
+    ObjectSpec createObject(String objectName, boolean useQuotedNames);
```
Comment (frant-hartm) [inline hazelcast/src/test/java/com/hazelcast/test/jdbc/GenericMapStoreObjectProvider.java:31]: Is `Object` the right name? Maybe `record` instead?

**Review by frant-hartm — CHANGES_REQUESTED** [review]

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapLoaderTest.java (hunk 2 above)
Comment (TomaszGaweda) [reply extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapLoaderTest.java]: Done :) Multiple revisions, so just please take a look overall

File: hazelcast/src/test/java/com/hazelcast/test/jdbc/GenericMapStoreObjectProvider.java (hunk 1 above)
Comment (TomaszGaweda) [reply hazelcast/src/test/java/com/hazelcast/test/jdbc/GenericMapStoreObjectProvider.java]: Done :) Multiple revisions, so just please take a look overall

**Comment by TomaszGaweda** [comment]
run-nightly-tests

**Comment by TomaszGaweda** [comment]
run-nightly-tests

**Comment by TomaszGaweda** [comment]
run-lab-run

**Comment by TomaszGaweda** [comment]
run-nightly-tests

**Pull request closed without merging**